src/libraries/System.Security.Cryptography/tests/X509Certificates/NameTests.cs

```diff
@@ -82,22 +83,52 @@ public static void ForceUtf8EncodingWithFlagWhenEncoding()
                 "CN=potato, O=jicama",
                 X500DistinguishedNameFlags.ForceUTF8Encoding);
 
-            ReadOnlySpan<byte> expectedDer = new byte[]
-            {
+            ReadOnlySpan<byte> expectedDer =
+            [
                 0x30, 0x22,
                 0x31, 0x0F,
                 0x30, 0x0D,
                     0x06, 0x03, 0x55, 0x04, 0x03, // id-at-commonName OID
                     0x0C, 0x06, 0x70, 0x6F, 0x74, 0x61, 0x74, 0x6F, // 0x0C is UTF8String
                 0x31, 0x0F,
                 0x30, 0x0D,
-                    0x06, 0x03, 0x55, 0x04, 0x0A,  // id-at-organizatioName OID
+                    0x06, 0x03, 0x55, 0x04, 0x0A,  // id-at-organizationName OID
                     0x0C, 0x06, 0x6A, 0x69, 0x63, 0x61, 0x6D, 0x61, // 0x0C is UTF8String
-            };
+            ];
 
             AssertExtensions.SequenceEqual(expectedDer, name.RawData);
         }
 
+        [Theory]
+        [InlineData("G=DotNet", UniversalTagNumber.UTF8String)]
+        [InlineData("L=Alexandria", UniversalTagNumber.UTF8String)]
+        [InlineData("O=GitHub", UniversalTagNumber.UTF8String)]
+        [InlineData("OU=ProdSec", UniversalTagNumber.UTF8String)]
+        [InlineData("S=Virginia", UniversalTagNumber.UTF8String)]
+        [InlineData("SN=Doe", UniversalTagNumber.UTF8String)]
+        [InlineData("ST=Main", UniversalTagNumber.UTF8String)]
+        [InlineData("T=Pancake", UniversalTagNumber.UTF8String)]
+        [InlineData("CN=Foo", UniversalTagNumber.UTF8String)]
+        [InlineData("E=noone@example.com", UniversalTagNumber.IA5String)]
+        [InlineData("OID.1.2.3.4=sample", UniversalTagNumber.PrintableString)]
+        [InlineData("C=US", UniversalTagNumber.PrintableString)]
+        public static void ForceUtf8EncodingForEligibleComponents(string distinguishedName, UniversalTagNumber tagNumber)
```

Comment:
`[InlineData("OID.2.5.4.43=VCS", UniversalTagNumber.UTF8String)]`

and maybe redundantly specify one of the others using the OID form, e.g. OU => OID.2.5.4.11